### PR TITLE
tests: Improve testing documentation

### PR DESCRIPTION
**Pull request by mxinden**
- Move testing documentation into `tests/README.md`
- Extend PR trigger documentation with triggering by platform label
- Remove outdated documentation on folder structure

### CODE REVIEW

File: tests/README.md [hunk 1]
```diff
@@ -1,17 +1,72 @@
...
-    │   └── vars   # Terraform tfvars files for AWS smoke tests
-    ├── azure      # Smoke tests for Azure
-    │   └── vars   # Terraform tfvars files for Azure smoke tests
-    ├── bare-metal # Smoke tests for bare-metal
-    │   └── vars   # Terraform tfvars files for bare-metal smoke tests
-    └── ...
+CLUSTER
+AWS_ACCESS_KEY_ID
+AWS_SECRET_ACCESS_KEY
+TF_VAR_tectonic_aws_ssh_key
```
Comment (cpanato) [inline tests/README.md]: this you don't need since we are generating random keys everytime

Comment (mxinden) [reply tests/README.md]: Thanks, fixed.

File: tests/README.md [hunk 2]
```diff
@@ -1,17 +1,85 @@
...
+  executions. The result of the most recent execution will be reported as a PR
+  status in GitHub.
+
+- *What can I do in case I run into test flakes continually?*
+
+  1. Make sure the test failure is in **no** way related to your PR changes.
+     Test your changes locally thoroughly.
+  2. Document the flake in Jira in the `INST` project as *issue type* "bug" with the
+     `flake` label.
+  3. Merge the PR.
```
Comment (cpanato) [inline tests/README.md]: `Merge the PR if have approvals`

Comment (cpanato) [reply tests/README.md]: or something like that

File: tests/README.md [hunk 3]
```diff
@@ -1,17 +1,86 @@
...
+### FAQ
+- *How do I retrigger the tests?*
+
+  Comment with `ok to test` on the PR.
+
+
+## Running smoke tests on PRs
+
+In addition to our basic set of tests we have smoke tests. These test the
+Tectonic installer on our supported platforms.
```
Comment (squat) [inline tests/README.md]: add a colon after platforms to introduce the list.

File: tests/README.md [hunk 4]
```diff
@@ -1,17 +1,86 @@
...
+CLUSTER
+AWS_ACCESS_KEY_ID
+AWS_SECRET_ACCESS_KEY
+TF_VAR_tectonic_aws_region
+TF_VAR_tectonic_license_path
+TF_VAR_tectonic_pull_secret_path
+TF_VAR_base_domain
 ```
+
+> Make sure both the *Tectonic pull secret* as well as the *Tectonic license* is
```
Comment (squat) [inline tests/README.md]: I don't think this paragraph is strictly true. The Make target for `tests/smoke` automatically bind mounts several directories including the license and pull secret paths:
```
	docker run \
	--rm \
	-it \
	-v "${CURDIR}":"${CURDIR}" \
	-w "${CURDIR}/tests/rspec" \
	-v "${TF_VAR_tectonic_license_path}":"${TF_VAR_tectonic_license_path}" \
	-v "${TF_VAR_tectonic_pull_secret_path}":"${TF_VAR_tectonic_pull_secret_path}" \
	-v "${SSH_AUTH_SOCK}:${SSH_AUTH_SOCK}" \
	-v "${TF_VAR_tectonic_azure_ssh_key}":"${TF_VAR_tectonic_azure_ssh_key}" \
...
```

**Pull request merged**